Lib/fontTools/ttLib/ttFont.py

```diff
@@ -666,7 +666,17 @@ def getGlyphNameMany(self, lst):
     def getGlyphID(self, glyphName):
         """Returns the ID of the glyph with the given name."""
         try:
-            return self.getReverseGlyphMap()[glyphName]
+            gid = None
+            try:
+                gid = self.getReverseGlyphMap()[glyphName]
```

Comment:
I think you just return directly here, in the happy case, and not bother with the additional `self.getGlyphName(gid) == glyphName`. The glyf `__setitem__` only appends at the end of the glyphOrder when it's a _new_ glyphName, otherwise it overwrites an existing glyph in place and keeps the current glyphOrder, nothing ever 'moves' an existing named glyph. And glyf `__setitem__` is I think the only place where a TTFont.glyphOrder list is ever modified (or rather allowed to modify), apart from the TTFont.setGlyphOrder() itself. Supporting changing the glyph index of existing named glyphs is trickier than appending at the end because many glyph-related tables need modification, and I think we should not support this use case in the getGlyphID API. If one intentionally wishes to reshuffle glyphs, they should explicitly call font.setGlyphOrder() and rebuild or invalidate the cache (e.g. font.getReverseGlyphMap(rebuild=True) or deleting `_reversedGlyphOrderDict` attribute). Quicker/easier to assume that if a glyphName is present in the current `_reverseGlyphOrderDict` cache, then the cached index is the good one; otherwise if absent (i.e. the KeyError case) go on to rebuild the cache.

Reply:
> glyf __setitem__ is I think the only place where a TTFont.glyphOrder list is ever modified (or rather allowed to modify)

oh.. actually that's not true unfortunately. In addition to `__setitem__`, the glyf table object also has a `__delitem__` which removes from the glyphOrder without updating the TTFont's `_reversedGlyphOrderDict`, basically screwing the glyph index of all the glyphs that follows the removed glyph... :(